jenkins/development.Dockerfile

```diff
@@ -60,60 +65,131 @@ RUN \
+        python3-redis \
+        python3-requests \
+        python3-sh \
+        redis \
+        rpmdevtools \
+        rsync \
+        screen \
+        sos \
+        tar \
+        xz
+
+# Install Server dependencies
+#
+# Note, this list is manually generated via `dnf -q repoquery --requires <RPM>`
+# targeting the Server RPM -- any dependencies missing from here should
+# presumably be added there, first.
+RUN \
+    dnf install -y \
+        npm \
+        policycoreutils \
+        policycoreutils-python-utils \
+        python3 \
+        python3-alembic \
+        python3-aniso8601 \
+        python3-bcrypt \
+        python3-boto3 \
+        python3-certifi \
+        python3-click \
+        python3-dateutil \
+        python3-devel \
         python3-elasticsearch \
-        python3-entrypoints \
-        python3-flake8 \
+        python3-email-validator \
         python3-flask \
         python3-flask-cors \
+        python3-flask-httpauth \
+        python3-flask-migrate \
         python3-flask-restful \
-        python3-gitdb \
+        python3-flask-sqlalchemy \
+        python3-greenlet \
         python3-gunicorn \
         python3-humanize \
-        python3-itsdangerous \
-        python3-jinja2 \
-        python3-jinja2-cli \
         python3-libselinux \
-        python3-mock \
-        python3-pathspec \
-        python3-pip \
-        python3-psutil \
         python3-psycopg2 \
+        python3-requests \
+        python3-sqlalchemy \
+        python3-sqlalchemy-utils
+
+# Install common lint and testing dependencies
+#
+# Note that the versions of Black and Flake8 are almost certainly wrong, but,
+# if so, the setup will install the proper versions.  The packages in this list
+# other than Black and Flake8 cover their dependencies when installed via pip.
+RUN \
+    dnf install -y \
+        black \
+        python3-flake8 \
+        python3-toml \
+        python3-regex \
+        python3-typed_ast \
+        python3-pathspec \
+        python3-attrs \
+        python3-appdirs \
+        python3-entrypoints \
+        python3-pycodestyle \
+        python3-mccabe \
+        python3-pyflakes
+
+# Install Agent testing dependencies
+#
+# Note that the last two packages are dependencies of the pip-installed PyTest
+# package; when we switch to the RPM-installed version, we should drop them.
+RUN \
+    dnf install -y \
+        python3-coverage \
+        python3-GitPython \
+        python3-mock \
         python3-pytest \
         python3-pytest-cov \
         python3-pytest-helpers-namespace \
         python3-pytest-mock \
-        python3-redis \
-        python3-regex \
-        python3-requests \
-        python3-requests-mock \
         python3-responses \
-        python3-s3transfer \
-        python3-sh \
-        python3-smmap \
-        python3-sphinx \
-        python3-tox \
-        python3-tox-current-env \
-        python3-typed_ast \
+        \
+        python3-atomicwrites \
```

Comment:
Should this be in alphabetical order? Or is the "almost blank line" intended to separate this into sections? And, if so, can you document the significance of these sections?